### PR TITLE
Support new Azure audit logs and signin logs

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to add support for the new Azure audit logs and signin logs. Sample logs are provided by the Azure team and added into `/_dev/test/pipeline` for both `auditlogs` data stream and `signinlogs` data stream in Azure package. 

This PR added two fields for audit logs and 4 fields for signin logs:

- azure.auditlogs.level
- azure.auditlogs.properties.authentication_protocol
- azure.signinlogs.properties.authentication_protocol
- azure.signinlogs.properties.incoming_token_type
- azure.signinlogs.properties.service_principal_credential_key_id
- azure.signinlogs.properties.unique_token_identifier

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/azure/data_stream/signinlogs/fields/fields.yml [hunk 1]
```diff
@@ -149,7 +149,7 @@
             - name: trust_type
               type: keyword
               description: |
-                Trust type
+                Trust type\
```
Comment (aspacca) [inline packages/azure/data_stream/signinlogs/fields/fields.yml:152]: trailing `\`

File: packages/azure/data_stream/signinlogs/fields/fields.yml [hunk 2]
```diff
@@ -206,3 +206,15 @@
           type: boolean
         - name: sso_extension_version
           type: keyword
+        - name: authentication_protocol
```
Comment (aspacca) [inline packages/azure/data_stream/signinlogs/fields/fields.yml:209]: where these new fields are extracted?

File: packages/azure/docs/README.md [hunk 3]
```diff
@@ -804,10 +808,11 @@ An example event for `signinlogs` looks as following:
 | azure.signinlogs.properties.device_detail.device_id | Device ID | keyword |
 | azure.signinlogs.properties.device_detail.display_name | Display name | keyword |
 | azure.signinlogs.properties.device_detail.operating_system | Operating system | keyword |
-| azure.signinlogs.properties.device_detail.trust_type | Trust type | keyword |
+| azure.signinlogs.properties.device_detail.trust_type | Trust type\ | keyword |
```
Comment (aspacca) [inline packages/azure/docs/README.md:811]: trailing `\`

File: packages/azure/data_stream/signinlogs/fields/fields.yml (hunk 2 above)
Comment (kaiyan-sheng) [reply packages/azure/data_stream/signinlogs/fields/fields.yml:209]: @aspacca These new fields are from the new sample logs we got from Azure. Based on the new sample logs, we are adding support to these new fields.

Comment (aspacca) [reply packages/azure/data_stream/signinlogs/fields/fields.yml:209]: @kaiyan-sheng , sorry, I didn't explain myself precisely:
I see the fields are camelCase (`authenticationProtocol`, `incomingTokenType`) while we map them as snake_case, and I cannot find where we convert the case :)

Comment (aspacca) [reply packages/azure/data_stream/signinlogs/fields/fields.yml:209]: @kaiyan-sheng , sorry, I didn't explain myself precisely:
I see the fields are camelCase (`authenticationProtocol`, `incomingTokenType`) while we map them as snake_case, and I cannot find where we convert the case :)

Comment (kaiyan-sheng) [reply packages/azure/data_stream/signinlogs/fields/fields.yml:209]: @aspacca Ahh this code is in Beats I think: https://github.com/elastic/beats/blob/main/x-pack/filebeat/module/azure/signinlogs/ingest/pipeline.yml#L15

**Review by aspacca — APPROVED** [review]

**Pull request merged**